Event

sendilkumarn opened the pull request.

sendilkumarn commented:
fixes #1887

annevk commented:
Thank you for writing this patch! I noticed a couple of issues:

* To link the term "fire an event" to its definition, one needs to write `<span data-x="concept-event-fire">fire an event</span>`. You could change that around so that you only need to write `<span>fire an event</span>`, but then you'd have to update existing references and the `<dfn>`.
* It seems like you did a case-insensitive search & replace. Sometimes "Fire" needs to be uppercase. And also, because you did that I suspect not everything is 100 column wrapped anywhere, per the description at https://github.com/whatwg/html/blob/master/CONTRIBUTING.md#formatting.

Please do let me know if you have any questions.

sendilkumarn commented:
@annevk yeah I found that out. 
Can we add how to locally run test before submitting PR in `contributing` docs ?
So it would be helpful for new ones like me.

annevk commented:
To build locally you need to follow the instructions here: https://github.com/whatwg/html-build. It's mentioned in the CONTRIBUTING document, but maybe not clearly enough?

I think the issues I noted about around 100 column-line-length and "Fire a simple event" becoming "fire an event" instead of "Fire an event" are still there. So those would still need to be fixed before we can proceed.

annevk commented:
So I thought I'd do some of the wrapping for you, since it didn't all go according to the guidelines, but we also try to be a little more lenient with someone's first PR. However, it turns out you also wrapped things that this PR did not need to touch, such as
```diff
+     <li><p>Let <var>notCanceled</var> be the result of <span
+     data-x="concept-event-fire">firing an event</span> named <code
+     data-x="event-invalid">invalid</code> at <var>field</var>, with the <code
+     data-x="dom-Event-cancelable">cancelable</code> attribute initialized to true.</p></li>
```
(This is wrong as you can wrap wherever there is a space. Wrapping directly after "firing an" as the text used to do is correct.)

Could you undo that? Also, you can add your name to the acknowledgments.

annevk commented:
@sendilkumarn to be clear, I don't think that's the only line you affected. Did you do the changes manually or run some kind of script?

sendilkumarn commented:
@annevk yeah, I replaced all the `firing an event` reference.

>changes manually or run some kind of script? 

It was a script initially and then I switched to 50-50, find with script and replace manually. 

I will rebase now.

sendilkumarn commented:
@annevk thanks for the time and help. Please let me know if I have missed any other thing.

annevk commented:
I ended up fixing the remainder. You might want to look at those diffs to see what you missed and can improve upon in a future PR.

In any event, thanks so much for all this work, it's greatly appreciated and is a nice cleanup of the HTML Standard!

sendilkumarn commented:
Thanks 👍  will contribute more.
